### DISCUSSION TITLE
Marketplace

### DISCUSSION THREAD

**Discussion by luizbills** [4ian/GDevelop#2330] [Enhancement ideas (and their technical discussions)]
GDevelop could have a marketplace for free and paid game assets and editor extensions. Part of the sales would go to development. And I believe this would attract other artists and developers to our community and even help in promoting the game editor.

**Comment by luizbills** [comment]
Instead of charging a commission on sales, it could charge a monthly fee to sell premium items.

**Comment by 4ian** [comment]
Please rather use the forum for ideas/feature requests :) 
In theory yes that would be great to have a store/marketplace. In practice, making something that is really good is hard because you have to come up with: 
- The store interface/website,
- The upload/download/hosting
- The account management (we have accounts in GDevelop so we could reuse them, but have to see how to integrate them to the marketplace)
- The payment (kind of feasible with Stripe marketplace) 
- The legal things
- The moderation of content (do you moderate after upload, before making things available online, etc...) - which takes time in addition to the maintenance of the game engine and the software itself.
- The integration with the engine.

So while I would be all for it if we had the resources, for now we would a pragmatic solution or an integration with something already existing maybe?

**Comment by blurymind** [comment]
I quite like how itch.io does it :) you can put out stuff for free and people donate if they want to. You can also make it paid with minimal donation.

Why not integrate itch.io itself inside the newIde. Then it could handle accounts,payments, etc.
Itch has an open source app btw
https://github.com/itchio/itch

I wonder  if they have some sort of an api that can be used to use their services in more integrated ways.

Could we add an itchio section for gdevelop with subsections, then load that inside electron, then blam - most of the things are now handled by the itchio team and we can focus on gdevelop..

heck it would be great if people could publish from gd straight to itch
https://www.npmjs.com/package/itchy-electron

take full advantage of the fact we are running in electron :D 

The upside to itch is that it already has a huge formed community! It would be similar to embedding discort into the newide

**Comment by luizbills** [comment]
> Please rather use the forum for ideas/feature requests :)

Sorry for that.

> So while I would be all for it if we had the resources, for now we would a pragmatic solution or an integration with something already existing maybe?

I understood.

---

> Why not integrate itch.io itself inside the newIde.

The marketplace could help increase your revenue and help the project grow faster.

> Then it could handle accounts, payments, etc.

Yes, but it wouldn't help the project much. And if someone wanted to sell an extension, the buyer would have to download and import manually. 

Without an official marketplace, developers won't feel too confident about making paid extensions or paid tutorials.

---

Anyway, maybe in the future this might become a reality.

**Comment by blurymind** [comment]
itch is a legit place to sell content to develop games. It is already full of both 2d game assets, other tools and code for different game engines. It's where a lot of game jams happen. Its like the open source cousin of steam, but without the entry fees..

If you actually look at the content, and the level of quality of content - you will see that it is probably our best option :) not to burst your bubble, but do you want us to work hard on making the engine good or do you want us to spend our time maintaining and policing a closed market place that is exclusive only to gdevelop users? Doesn't seem like the best idea to me. Taking on an existing, already full of content market space is a much more exciting move than reinventing the wheel.

Straight from day one we give GD users access to that huge pool of content - no registration required, beautifully integrated in the ide. If they want to sell it, buy it, share it - the infrastructure is all there - battle tested and used by tons of other people, who use other engines. GD shared content at itch can be discovered by the many users of that site and therefor spark more interest in GD. If you are an artist, your shared content is usable in other game engines. If you share code - its GD exclusive.

VS

Recreate everything from scratch, then maintain it, hope people will use it. Have low quality content or very little content from day one, some of the GD users might use it, nobody else will find that content but GD users, which lowers the incentive to want to sell it at a platform that doesn't get that many visits. Most indie game content creators will use itch instead imo, or put very specific to GD content only. No graphics, no effects, no music. Just example project files

I say go to itch, because that is where the community is. That is where we will get more people to use GD. However you look at it, doing it ourselves from scratch is an uphill battle not worth fighting. We can gain way more from using an established marketplace
